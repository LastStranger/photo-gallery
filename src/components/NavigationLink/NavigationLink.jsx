// @flow

import { h, Component } from 'preact';
import { route } from 'preact-router';

class NavigationLink extends Component {
  static defaultProps = {
    delay: 1000,
    className: ''
  };
  props: {
    delay?: number,
    children: mixed,
    to: string,
    onStart: Function,
    className?: string
  };
  delayNavigation = (event: MouseEvent) => {
    event.stopPropagation();
    event.preventDefault();
    this.props.onStart();
    setTimeout(() => {
      route(this.props.to);
    }, this.props.delay);
    return false;
  };
  render() {
    return (
      <a className={this.props.className} href={this.props.to} onClick={this.delayNavigation}>
        {this.props.children}
      </a>
    );
  }
}

export default NavigationLink;
