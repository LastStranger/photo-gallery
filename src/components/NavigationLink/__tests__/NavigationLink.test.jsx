// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import NavigationLink from '../';

jest.useFakeTimers();

test('renders', () => {
  const string = render(
    <NavigationLink to="/test/link" className="test-class" onStart={() => {}}>Test Link</NavigationLink>
  );
  expect(string).toMatchSnapshot();
});

test('delays navigation', () => {
  const instance = new NavigationLink();
  const stopPropagation = jest.fn();
  const preventDefault = jest.fn();
  const onStart = jest.fn();
  instance.props = { onStart, to: '/more/url/things', children: 'more testing stuff' };
  instance.delayNavigation(
    ({
      stopPropagation,
      preventDefault,
      onStart
    }: any)
  );
  expect(stopPropagation).toBeCalled();
  expect(preventDefault).toBeCalled();
  expect(onStart).toBeCalled();
});
