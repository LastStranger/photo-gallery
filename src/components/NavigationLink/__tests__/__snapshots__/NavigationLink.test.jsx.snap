// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `<a class="test-class" href="/test/link">Test Link</a>`;
